client/components/notice/index.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+
+import React, { useEffect, FunctionComponent, useRef } from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+/**
+ * Constants
+ */
+enum Status {
```

Comment:
We already have existing types for the notice statuses - [see here](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/notices/types.ts#L16). 

Maybe they would be worth moving to a more generic place now that they'll be used for more than just state?

Also, a side note here is that I'd prefer using [union types instead of enums](https://blog.bam.tech/developer-news/should-you-use-enums-or-union-types-in-typescript) for those, but this is already done for `NoticeStatus` as you can see.